railties/test/generators/app_generator_test.rb

```diff
@@ -338,6 +338,18 @@ def test_app_update_preserves_skip_test
     end
   end
 
+  def test_app_update_preserves_skip_system_test
+    app_root = File.join(destination_root, "myapp")
+    run_generator [ app_root, "--skip-system-test" ]
+
+    FileUtils.cd(app_root) do
+      config = "config/application.rb"
+      assert_no_changes -> { File.readlines(config).grep(/generators\.system_tests/) } do
```

Comment:
This may be a bit paranoid, but if the mechanism for disabling system tests changes (e.g. the config key is renamed or the location of this generated config is changed), then this assertion will still pass (because `[]` == `[]`).

The other `test_app_update_preserves_skip_*` suffer from this problem too, but it feels like less of an issue with `require` statements.

So what do you think about adding an extra assertion?

```suggestion
      assert_file config, /generators\.system_tests/
      assert_no_changes -> { File.readlines(config).grep(/generators\.system_tests/) } do
```

Reply:
Makes sense to me, I agree this could easily slip by in the future

Reply:
Oops...  Based on the [build failure](https://buildkite.com/rails/rails/builds/88273#01823207-48f4-4f1e-a202-6bb00538fa9b/1214-2564), I think `assert_file config` needs to go outside of the `FileUtils.cd` block.

Reply:
I haven't dug into why these tests are generating inside `destination_root/myapp`, but since `assert_file` wants a relative_path from `destination_root` it appears that we have to add `myapp/` and the `FileUtils.cd` doesn't affect it either way https://github.com/rails/rails/blob/76b0ef8d00e3d824c42f3b71319edc15e7aa7a3e/railties/lib/rails/generators/testing/assertions.rb#L26

Reply:
Oh, I see.  It's because of `app_root = File.join(destination_root, "myapp")`.  I'm not sure about the purpose of that convention.  It seems to be unnecessary indirection, and I have [simply removed it](https://github.com/rails/rails/commit/b2c5199d5107129ff81c2e6f44e7f6bb87a3b94a#diff-9dd1c2e31fd5617506e344d6328e0e165614290478e1180ef0630017b9305ba4L192-R205) in the past.  But we can keep it for this test, to match the others.

Reply:
Okay, I blamed a bit and found the first test to introduce `app_root`: cdad483. My thinking is that this is specifically done to test against the application's derived name and is unnecessary for tests that don't care about naming. Would you accept a commit/new PR to remove `app_root`/`myapp` folder from any tests that don't need it?

Reply:
> I haven't dug into why these tests are generating inside `destination_root/myapp`

Hah, you beat me to it! :laughing:

Reply:
> Would you accept a commit/new PR to remove `app_root`/`myapp` folder from any tests that don't need it?

Sounds good! :+1: